src/go/k8s/controllers/redpanda/redpanda_controller.go

```diff
@@ -75,6 +75,7 @@ type RedpandaReconciler struct {
 // +kubebuilder:rbac:groups=apps,namespace=default,resources=replicasets,verbs=get;list;watch;create;update;patch;delete
 
 // additional k8s resources required by flux
+// +kubebuilder:rbac:groups="",namespace=default,resources=pods,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Commit message nor PR description explain why this is needed. `helm` does not create pods, right? Only Deployments, Jobs, Statefulset.

Reply:
There is a description of the change is on the preamble here.  Pods can be created by helm, for example look at tests. I cannot say for certain we will not need to create them in the future. We are strongly considering using namespace admin rights for flux to avoid having to continue supporting adding more rules to rbac.

Reply:
> here is a description of the change is on the preamble here.

It would be good to understand what workflow/test caused it to bubble up in commit message or PR cover letter. 

> Pods can be created by helm, for example look at tests.

I don't see any point in supporting helm test in operator deployment. It should be only scoped to helm itself, right?

>  We are strongly considering using namespace admin rights for flux to avoid having to continue supporting adding more rules to rbac.

The whole purpose of having namespaced operator was to narrow RBAC scope. Now we have added flux custom resource RBAC rules on top of what we had in operator v1. The only change is `ClusterRole` changed into `Role`.